### PR TITLE
JWT middleware

### PR DESCRIPTION
**Pull request by luizbafilho**
Closes: #635 

**What changed?**
- Adding middleware to validate JWT token and inject the provider token into the request's context.

**Why?**

**How did you test it?**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: pkg/middleware/middleware_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+		midware := middleware.WithProviderToken(jwtClient, defaultHandler)
+		req := httptest.NewRequest(http.MethodGet, "http://www.foo.com", nil)
+		req.Header.Add("Authorization", "token my-jwt-token")
+
+		res := httptest.NewRecorder()
+
+		midware.ServeHTTP(res, req)
+
+		Expect(res.Result().StatusCode).To(Equal(http.StatusForbidden))
```
Comment (jpellizzari) [inline pkg/middleware/middleware_test.go:69]: Nit: Would `http.StatusUnauthorized` be more appropriate?

**Review by jpellizzari — APPROVED** [review]
LGTM. Nice work on the test coverage.

File: pkg/middleware/middleware_test.go (hunk 1 above)
Comment (luizbafilho) [reply pkg/middleware/middleware_test.go:69]: yes, I'll change it

**Review by josecordaz — APPROVED** [review]
LGTM

**Pull request merged**